frontend/front/src/redux/adminViewApiSlice.js

```diff
@@ -1,12 +1,36 @@
 import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
 
-const baseUrl = "http://localhost:3001";
+// This custom base query function checks the availability of the local API
+// and the Vercel-hosted API. It sets the base URL according to the API
+// that is available, so you don't need to manually change the base URL.
+//
+// To access and visualize the data from the Vercel-hosted API, simply
+// visit the API endpoint URLs using the Vercel deployment domain
+// (e.g., https://https://bhpa.vercel.app/api/your-endpoint).
+async function customBaseQuery(args, api, extraOptions) {
+  const localApiUrl = "http://localhost:3001";
+  const vercelApiUrl = "https://bhpa.vercel.app/api";
+
+  // Check if the local API is available
+  const isLocalApiAvailable = await fetch(localApiUrl)
```

Comment:
Checking for the API like this seems kind of janky - because this is built into the baseQuery it's doing an extra request on EVERY request this api slice makes. IMO the better way to do this would be with an environment variable. Maybe Vercel has some sort of variable it makes available during the build process that you could use in the logic here to tell if its Vercel or not? Or you could go the other direction and use `process.env.NODE_ENV === "development"` to test if it was started with `yarn start`.

Reply:
Hi @mattdelsordo! Thanks for your feedback on the code. I agree that using an environment variable would be a better approach to checking if the API is available. I will look into the documentation for variables on Vercel and implement this change. Thanks again for your suggestion!